lib/ansible/executor/module_common.py

```diff
@@ -595,39 +595,45 @@ def _get_shebang(interpreter, task_vars, templar, args=tuple()):
        file rather than trust that we reformatted what they already have
        correctly.
     """
-    interpreter_name = os.path.basename(interpreter).strip()
-
     # FUTURE: add logical equivalence for python3 in the case of py3-only modules
 
-    # check for first-class interpreter config
+    interpreter_name = os.path.basename(interpreter).strip()
+
+    # name for interpreter var
+    interpreter_config = u'ansible_%s_interpreter' % interpreter_name
+    # key for config
     interpreter_config_key = "INTERPRETER_%s" % interpreter_name.upper()
 
-    if C.config.get_configuration_definitions().get(interpreter_config_key):
-        # a config def exists for this interpreter type; consult config for the value
-        interpreter_out = C.config.get_config_value(interpreter_config_key, variables=task_vars)
-        discovered_interpreter_config = u'discovered_interpreter_%s' % interpreter_name
+    # skip detection for network os execution, use playbook supplied one if possible
+    if task_vars.get('ansible_network_os') and task_vars.get('ansible_playbook_%s' % interpreter_name):
```

Comment:
Don't we only want this case to fire specifically for Python? I don't think it's a problem in the real world today, but with `ansible_playbook_X`, I don't think X would ever be defined for anything other than `python`, right?

Reply:
currently yes, only python has such var, but thinking in case we expand in future to python3 (hopefully never ruby/perl/etc)